pkg/cert/request.go

```diff
@@ -65,7 +64,9 @@ func NewServerCertificateManager(kubeClient clientset.Interface, namespace, secr
 			// authenticate itself to a TLS client.
 			certificates.UsageServerAuth,
 		},
-		SignerName:          certificates.LegacyUnknownSignerName,
+		// Hard coding this since LegacyUnknownSignerName is no longer available in certificates/v1
+		// https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers.
+		SignerName:          "kubernetes.io/legacy-unknown",
```

Comment:
it's not just unavailable, it's not allowed: https://github.com/kubernetes/kubernetes/blob/48da959dbff18bfef6e801bd8c8ab3c88b7a7650/pkg/apis/certificates/validation/validation.go#L202

I think we will need to deprecate in-cluster flag + print a clear error + exit gracefully if the kubernetes server version is 1.22 AND `in-cluster` is true. Instead of trying and failing to create invalid CSR. However, doing all that is out of scope of your PR. So I would like to accept the other changes and drop these changes to ` pkg/cert/request.go`, if possible.

Reply:
Thanks for the feedback @wongma7, and it totally makes sense to separate these. The only reason I changed `pkg/cert/request.go` is because of dependencies. `certwatcher` requires a newer version of the `controller-runtime`, which resulted in a upgrade of `k8s.io/client-go` to `v0.23.0`, which is turn changed the underlying [Config type](https://pkg.go.dev/k8s.io/client-go/util/certificate#Config) for `KeyUsage` to `certficates/v1`.

I'm not entirely sure if there is a way to avoid these dependency updates for now, but would be open to any suggesstions.

Reply:
I see, thx for the explanation. Then I agree with merging this part as well, I don't see a way to avoid it either.

Reply:
Note for self and others interested in the `in-cluster=true` case:

Given that this PR basically solves the `in-cluster=false` case since now webhook actually reload certs when they are renewed...

My plan for `in-cluster=true` is...

1. Currently, `in-cluster=true` works for k8s <=1.21 but doesn't for >1.21. We have discussed this at length and there seems to be no way to make it work in >1.22 (without mis/abusing the CSR api) so we need to deprecate it ASAP (as 1.22 has been out for months now).
2. Upon merging this, `in-cluster=true` won't work for ANY k8s version. Users who rely on it will need to make sure they use an older revision when building pod-identity-webhook.
3. Immediately/soon after merging this, someone (me/colleague/contributor) should deprecate `in-cluster=true` with a note in README and major version release so it is clear & obvious that it won't work for ANY k8s version

Reply:
> 2\. Upon merging this, `in-cluster=true` won't work for ANY k8s version. Users who rely on it will need to make sure they use an older revision when building pod-identity-webhook.

It probably makes sense to tag a release before merging this since the latest release is a bit old. Then once the followup work is done, a new release can be cut that introduces `certwatcher` and deprecates the `--in-cluster` flag.